src/v0/destinations/ga4/transform.js

```diff
@@ -43,31 +42,13 @@ const {
 /**
  * returns client_id
  * @param {*} message
- * @param {*} clientIdFieldIdentifier
  * @returns
  */
-const getGA4ClientId = (message, Config) => {
-  let clientId;
-
-  // if hybrid mode enabled, take client_id from integrationsObj
-  if (isHybridModeEnabled(Config)) {
-    const integrationsObj = getIntegrationsObj(message, 'ga4');
-    if (integrationsObj && integrationsObj.clientId) {
-      return integrationsObj.clientId;
-    }
-  }
-
-  // for cloud mode first we will search from webapp
-  if (Config.clientIdFieldIdentifier) {
-    clientId = get(message, Config.clientIdFieldIdentifier);
-  }
-  // if we don't find it from the config then we will fall back to the default search
-  if (!clientId) {
-    clientId =
-      getDestinationExternalID(message, 'ga4ClientId') ||
-      get(message, 'anonymousId') ||
-      get(message, 'messageId');
-  }
+const getGA4ClientId = (message) => {
+  const clientId =
+    getDestinationExternalID(message, 'ga4ClientId') ||
+    get(message, 'anonymousId') ||
+    get(message, 'messageId');
```

Comment:
Apart from messageId can't we use rudderId? MessageId doesn't serve the purpose of clientID at all as they are always different for all of the events which can be used for event deduplication but this is not the purpose of clientId isn't it?

Reply:
@ItsSudip this is a fallback option, ideally we are expecting anonymousId only or client_id inside externalId. but we can discuss this option with amy in tomorrow meeting and will post updates here

Reply:
did we discuss on this?